lib/multiplexer/multiplexer.go

```diff
@@ -414,75 +433,102 @@ func (m *Mux) detect(conn net.Conn) (*Conn, error) {
 
 		switch proto {
 		case ProtoProxy:
-			if !m.EnableExternalProxyProtocol {
+			newPROXYLine, err := ReadProxyLine(reader)
+			if err != nil {
+				return nil, trace.Wrap(err, invalidProxyLineError)
+			}
+
+			if m.PROXYProtocolMode == PROXYProtocolOff {
 				return nil, trace.BadParameter(proxyProtocolDisabledError)
 			}
-			// We allow only one unsigned proxy line
-			if proxyLine != nil {
-				return nil, trace.BadParameter(duplicateProxyLineError)
+
+			unsignedPROXYLineReceived = true
+
+			if m.PROXYProtocolMode == PROXYProtocolUnspecified {
+				m.Logger.WithFields(log.Fields{
+					"direct_src_addr": conn.RemoteAddr(),
+					"direct_dst_addr": conn.LocalAddr(),
+					"proxy_src_addr:": newPROXYLine.Source.String(),
+					"proxy_dst_addr:": newPROXYLine.Destination.String(),
+				}).Error(unexpectedPROXYLine)
```

Comment:
Can you improve the message so it's easier to follow?
Can you please add an actionable message here so users can easily fix their configs?

Reply:
I tweaked it a little bit, but overall I don't see much problem with the message and it already include actionable part?